src/pages/studyView/table/GeneTable.tsx

```diff
@@ -399,88 +423,49 @@ export class GeneTable extends React.Component<GeneTableProps, {}> {
         return !!this.props.cancerGeneFilterEnabled && this.props.filterByCancerGenes;
     }
 
+    @computed get allSelectedRowsKeys() {
+        return [...this.selectedRowsKeys, ...this.preSelectedRowsKeys];
+    }
+
     @autobind
     isChecked(uniqueKey: string) {
-        return rowIsChecked(uniqueKey, this.preSelectedRows, this.selectedRows);
+        return  _.some(this.allSelectedRowsKeys, (key) => key === uniqueKey);
```

Comment:
is there a possibility that there will be very many selected rows? if so, then I think we should use a map/set so that checking if uniqueKey is included is constant time